File: COMPILADOR/kokoslan_/src/src/java/ast/KoKoAtom.java
/**
 @author loriacarlos@gmail.com
 
*/
package src.java.ast;
import java.io.*;
import src.java.eval.KoKoContext;
import src.java.eval.KoKoValue;
import src.java.exception.KoKoEvalException;


public class KoKoAtom<T> implements KoKoAst{
   private T value;
   public T getValue(){return this.value;}
   
   public KoKoAtom(T value){
      this.value = value;
   }
   @Override
   public void genCode(PrintStream out){
      out.print(this.value + " ");
   }
   public KoKoValue eval(KoKoContext ctx){
	   throw new KoKoEvalException("KoKoAtom: eval not implemented");
	   
   }
   public String toString(){
	   return value.toString();
   }
}

File: COMPILADOR/kokoslan/src/KoKoKotlin/ast/KoKoAtom.kt
/**
 * @author loriacarlos@gmail.com
 */
package KoKoKotlin.ast

import KoKoKotlin.eval.KoKoContext
import KoKoKotlin.eval.KoKoValue
import KoKoKotlin.exception.KoKoEvalException
import src.java.ast.KoKoAst
import java.io.PrintStream


open class KoKoAtom<T>(val value: T) : KoKoAst {
    override fun genCode(out: PrintStream) {
        out.print(this.value.toString() + " ")
    }

    override fun eval(ctx: KoKoContext): KoKoValue {
        throw KoKoEvalException("KoKoAtom: eval not implemented")

    }

    override fun toString(): String {
        return value.toString()
    }
}